service/routes.py

```diff
@@ -230,3 +230,41 @@ def get_addresses(wishlist_id, id):
         )
 
     return jsonify(item.serialize()), status.HTTP_200_OK
+
+######################################################################
+# UPDATE WISHLIST ITEMS
+######################################################################
+@app.route("/wishlists/<int:wishlist_id>/items/<int:item_id>", methods=["PUT"])
+def update_items(wishlist_id, item_id):
+    
+    """
+    Update a Wishlist Item
+    This endpoint will update a wishlist item based the body that is posted
+    """
+    app.logger.info(
+        "Request to update a wishlist item %s for Wishlist id: %s",
+        (item_id, wishlist_id),
+    )
+    check_content_type("application/json")
+    wishlist = Wishlist.find(wishlist_id)
+    # see if the wishlist exists and abort if it doesn't
+    if not wishlist:
+        abort(
+            status.HTTP_404_NOT_FOUND,
+            f"Wishlist with id '{wishlist_id}' could not be found.",
+        )
+    # See if the item exists and abort if it doesn't
+    item = WishListItem.find(item_id)
+    if not item:
+        abort(
+            status.HTTP_404_NOT_FOUND,
+            f"Item with id '{item_id}' could not be found in the Wishlist with id '{wishlist_id}'.",
```

Comment:
There should be an additional check to see whether item id provided belongs to the specified wishlist or not. Say I make a PUT API call for {BASE_URL}/wishlists/1/items/2 but item id=2 belongs to wishlist id=2 and not wishlist id=1, then I should receive an error saying item id=2 not found for wishlist 1. You can check the read API call for reference

Reply:
I have updated the test case in the code.